interp/interp.go

```diff
@@ -314,6 +321,28 @@ func (interp *Interpreter) Eval(src string) (reflect.Value, error) {
 	return res, err
 }
 
+// Stop stops the execution of the interpreter
+func (interp *Interpreter) Stop() { interp.runid++ }
```

Comment:
This needs to update an atomic or be locked somehow. In general, people will want to run things in goroutines, but even here as this is implemented there is a race between the write to `interp.runid` here and the read in the `run` method.

I would suggest having this method unexported and have the user interact with it via a context as stated elsewhere.